### PR TITLE
Add file list upload support

### LINKED ISSUE #766: File upload list is not supported
When posting according to the spec https://github.com/jaydenseric/graphql-multipart-request-spec#curl-request-1 an error is raised.

uploadimage.sh
```
curl localhost:8000/graphql/ \
  -F operations='{ "query": "mutation ($files: [Upload!]!) { uploadMedia(files: $files) { ok } }", "variables": { "files": [null, null] } }' \
  -F map='{ "0": ["variables.files.0"], "1": ["variables.files.1"] }' \
  -F 0=@uploadimage.sh \
  -F 1=@uploadimage.sh
```

Mutation
```
@strawberry.type
class UploadSuccess:
    ok: bool = True


@strawberry.type
class Mutation:
    @strawberry.field
    def upload_media(self, files: List[Upload]) -> UploadSuccess:
        return UploadSuccess()
```

Result:
```
strawberry/file_uploads/data.py:31: in replace_placeholders
TypeError: list indices must be integers or slices, not str
```

### PR DESCRIPTION
**Pull request by DoctorJohn**
## Description

This PR adds support for uploading lists of files (`[Upload!]!`) to all integrations.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Documentation

## Issues Fixed or Closed by This PR

* Fixes #766 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)

File: strawberry/file_uploads/utils.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import copy
+from typing import Any, Dict, List, Mapping
+
+
+def replace_placeholders_with_files(
+    operations_with_placeholders: Dict[str, Any],
+    files_map: Dict[str, List[str]],
+    files: Mapping[str, Any],
+) -> Dict[str, Any]:
```
Comment (BryceBeagle) [inline strawberry/file_uploads/utils.py:9]:
```suggestion
def replace_placeholders_with_files(
    operations_with_placeholders: Dict[str, Any],
    files_map: Mapping[str, List[str]],
    files: Mapping[str, Any],
) -> Dict[str, Any]:
```

**Review by BryceBeagle — CHANGES_REQUESTED** [review]
This looks great :+1:

**Review by BryceBeagle — APPROVED** [review]

File: docs/guides/file-upload.md [hunk 2]
```diff
@@ -15,35 +17,47 @@ Since ASGI uses asyncio for communication the resolver _must_ be async as well.
...
+    async def read_files(self, files: typing.List[Upload]) -> typing.List[str]:
+        contents = []
+        for file in files:
+            content = (await file.read()).decode()
+            contents.append(content)
+        return contents
 ```
 
-## Flask
+## Sanic / Flask / Django / AIOHTTP
```
Comment (patrick91) [inline docs/guides/file-upload.md:40]: 💯

**Review by patrick91 — APPROVED** [review]
Wow, this looks great! I'll merge it now 😊

**Pull request merged**